src/Database/Concerns/BelongsToTenant.php

```diff
@@ -17,6 +17,9 @@ trait BelongsToTenant
 {
     use FillsCurrentTenant;
 
+    /**
+     * @return BelongsTo<\Stancl\Tenancy\Contracts\Tenant, $this>
```

Comment:
This won't work, all the relationships expect a Model, and this is a  contract. You will have to intersect them:


```suggestion
     * @return BelongsTo<\Illuminate\Database\Eloquent\Model&\Stancl\Tenancy\Contracts\Tenant, $this>
```